File: src/main/java/com/skysoft/mixin/ServerPackDataAccessor.java
package com.skysoft.mixin;

import com.google.common.hash.HashCode;
import java.util.UUID;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;

@Mixin(targets = "net.minecraft.client.resources.server.ServerPackManager$ServerPackData")
public interface ServerPackDataAccessor {
    @Accessor("id")
    UUID skysoft$getId();

    @Accessor("hash")
    HashCode skysoft$getHash();
}

File: src/main/kotlin/com/skysoft/mixin/ServerPackDataAccessor.kt
package com.skysoft.mixin

import com.google.common.hash.HashCode
import java.util.UUID
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.gen.Accessor

@Mixin(targets = ["net.minecraft.client.resources.server.ServerPackManager\$ServerPackData"])
interface ServerPackDataAccessor {
    @Accessor("id")
    fun skysoftGetId(): UUID

    @Accessor("hash")
    fun skysoftGetHash(): HashCode
}
